data-science-onramp/data-cleaning/clean.py

```diff
@@ -0,0 +1,190 @@
+import sys
+import re
+import datetime
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import UserDefinedFunction
+from pyspark.sql.types import StringType, IntegerType, FloatType
+
+
+PROJECT_ID = sys.argv[1]
+BUCKET_NAME = sys.argv[2]
+TABLE = f'{PROJECT_ID}.new_york_citibike_trips.RAW_DATA'
+
+def trip_duration(duration):
+    '''Convert trip duration to seconds.'''
+    if not duration:
+        return None
+    
+    time = re.match('\d*.\d', duration)
+
+    if not time:
```

Comment:
Should this error out? Or provide any kind of logging info? Or is it appropriate to just ignore? (I'm asking not to be passive aggressive - I'm asking so i can make sure I genuinely understand why things are happening the way they are)

Reply:
This case should not happen often. The format of the values in this column are either floats or ints, optionally followed by one of 's', 'm', or 'h'. This regex extracts the numerical part of the value, and then transforms it depending on the unit. It will fail to match (i.e. `not time` will be true) if there is no numerical value in the column, in which case it cannot be transformed. To my knowledge, the dirty dataset doesn't contain anything that will cause this to occur, but the following lines will cause runtime errors if it does occur, so it is there as a preventative measure.

Reply:
And to answer your question of returning None throughout - that only happens when some data is encountered for which we don't know how to transform/clean (e.g. a negative time). For these cases, we just want to eliminate the value altogether. Maybe there is value in printing something when this happens. But my thought is that generally this is what we want, and also this will happen so often for so many nonsensical values that if there was something of value, it would be difficult to find. What do you think?

Reply:
I'm happy with this solution as long as we add a function level comment explaining what gets returned when it's successful and what gets returned when it's not